plugins/palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/tasks/monitor_alerts/task.py

```diff
@@ -0,0 +1,381 @@
+                f"A PluginException has occurred. Status code 500 returned. Error: {error.cause}. "
+                f"Existing state: {existing_state}"
+            )
+            return [], existing_state, False, 500, PluginException(cause=error.cause, data=error)
+
+        except Exception as error:
+            print(f"{error = }")
+            return [], existing_state, False, 500, PluginException(preset=PluginException.Preset.UNKNOWN, data=error)
+
+    ###########################
+    # Make request
+    ###########################
+    def get_alerts_palo_alto(self, state, custom_config):
+        endpoint = "/public_api/v1/alerts/get_alerts"
+        response_alerts_field = "alerts"
+        time_sort_field = "creation_time"
+
+        self.logger.info(f"{ALERT_LIMIT = }")
+
+        # IT GOES 0 -> 100 | 101 -> 200 | 201 -> 300
+        # IF NULL, SET FROM = 0
+
+        if not state.get(LAST_SEARCH_FROM):
+            search_from = 0
+        else:
+            # STARTS 1 RECORD AFTER THE LAST ONE PULLED
+            search_from = state.get(LAST_SEARCH_TO) + 1
+            self.logger.info(f"{search_from = }")
+
+        # IF NULL SET TO = 100
+        if not state.get(LAST_SEARCH_TO):
+            search_to = 100
+        else:
+            # SEARCH_TO GETS THE NEWEST SEARCH FROM VALUE AND WILL STOP AFTER PULLING 99 ADDITIONAL RECORDS (100 TOTAL)
+            search_to = search_from + 99
+
+        state[LAST_SEARCH_FROM] = search_from
+        state[LAST_SEARCH_TO] = search_to
+
+        # search_from = state.get(LAST_SEARCH_FROM, 0)
+        # search_to = state.get(LAST_SEARCH_TO, 100) + 100
+
+        # search_to = search_from + ALERT_LIMIT
+        headers = self.connection.xdr_api.get_headers()
+
+        post_body = {
+            "request_data": {
+                "search_from": search_from,
+                "search_to": search_to,
+                "sort": {"field": time_sort_field, "keyword": "asc"},
+            }
+        }
+
+        url = urllib.parse.urljoin("https://api-rapid7.xdr.us.paloaltonetworks.com", endpoint)
+
+        # Local debugging
+        self.logger.info(f"State in UpDate: {state = }")
+        self.logger.info(f"{search_from = }")
+        self.logger.info(f"{search_to = }")
+        self.logger.info(f"{headers = }")
+        self.logger.info(f"{post_body = }")
+        self.logger.info(f"{url = }")
+
+        try:
+            request = requests.Request(method="post", url=url, headers=headers, json=post_body)
+            response = make_request(
+                request, exception_custom_configs=custom_config, timeout=120, allowed_status_codes=[]
+            )
+
+            response = response.json()
+
+            total_count = response.get("reply", {}).get("total_count", -1)
+            result_count = response.get("reply", {}).get("result_count", -1)
+            results = response.get("reply", {}).get(response_alerts_field, [])
+
+            is_paginating = ALERT_LIMIT < total_count
```

Comment:
I think this should be based on whether our result_count is greater or equal to 100 rather than total_count as total_count will always return a static number of results whether we have reached the end or not.